test/integration/EndToEnd.t.sol

```diff
@@ -654,13 +663,28 @@ contract EndToEndFlows is EndToEndUtils {
         uint128 amount
     ) internal {
         vm.startPrank(poolManager);
-        uint32 depositEpochId = hub.shareClassManager.nowDepositEpoch(shareClassId, assetId);
-        hub.hub.approveDeposits(poolId, shareClassId, assetId, depositEpochId, amount);
+        uint32 depositEpochId = hub.hubRequestManager.nowDepositEpoch(shareClassId, assetId);
+        D18 pricePoolPerAsset = hub.hubHelpers.pricePoolPerAsset(poolId, shareClassId, assetId);
+        hub.hub.callRequestManager(
+            poolId,
+            assetId.centrifugeId(),
+            abi.encodeCall(
+                IHubRequestManager.approveDeposits,
+                (poolId, shareClassId, assetId, depositEpochId, amount, pricePoolPerAsset)
+            )
+        );
```

Comment:
Because these calls seem complex and error prone with the encoding arguments, why not allow in `IHubRequestManager.approveDeposits` to be called by me FM directly?

approveDeposits could check: 
```
require(`msg.sender == hub || hub.isManager(msg.sender))
```
or
```
require(`wards[msg.sender] == 1 || hub.isManager(msg.sender))
```

It means, you can add it to the multicall if used through `hub.callRequestManager()` or you can use it directly by just calling `hubRequestDeposit.approveDeposits()`

Reply:
Hm, I see the advantage of having typed calls to the `hubRequestManager`, but I also think the downside is that it creates two paths to calling these methods, which is less than ideal. Will think about it more.

Reply:
Let's consider this in a followup PR.